api/v1alpha1/cachefsruntime_types.go

```diff
@@ -0,0 +1,275 @@
+	// instance will be used.
+	// E,g.
+	//   endpoint:
+	//     metaurl: "redis://instance-name.redis.rds.aliyuncs.com:6379/1"
+	// Or
+	//   endpoint:
+	//     encryptOptions:
+	//     - name: metaurl
+	//       valueFrom:
+	//         secretKeyRef:
+	//           name: cfs-secret
+	//           key: metaurl
+	//
+	// +optional
+	ExternalEndpoint EndpointSpec `json:"endpoint,omitempty"`
+}
+
+// EndpointSpec defines the configurations for external etcd cluster
+type EndpointSpec struct {
+	// MetaUrl specifies the endpoint of external master instance
+	// e,g. "redis://host:6379/1"
+	// +optional
+	MetaUrl string `json:"metaUrl,omitempty"`
+
+	// encrypted info for accessing the external master instance, only accept "metaurl" option
+	// name and this will override `MetaUrl` above, if both are specified.
+	// +optional
+	EncryptOptions []EncryptOption `json:"encryptOptions,omitempty"`
+}
+
+// CacheFSCompSpec is a description of the CacheFS components
+type CacheFSCompSpec struct {
+	// Ports used by CacheFS
+	// +optional
+	Ports []corev1.ContainerPort `json:"ports,omitempty"`
+
+	// Resources that will be requested by the CacheFS component.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Options
+	Options map[string]string `json:"options,omitempty"`
+
+	// Environment variables that will be used by CacheFS component.
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// NodeSelector is a selector
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// VolumeMounts specifies the volumes listed in ".spec.volumes" to mount into runtime component's filesystem.
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+
+	// PodMetadata defines labels and annotations that will be propagated to CacheFS's pods.
+	// +optional
+	PodMetadata PodMetadata `json:"podMetadata,omitempty"`
+
+	// Whether to use hostnetwork or not
+	// +kubebuilder:validation:Enum=HostNetwork;"";ContainerNetwork
+	// +optional
+	NetworkMode NetworkMode `json:"networkMode,omitempty"`
+}
+
+type CacheFSFuseSpec struct {
+	// Image for CacheFS fuse
+	Image string `json:"image,omitempty"`
+
+	// Image for CacheFS fuse
+	ImageTag string `json:"imageTag,omitempty"`
+
+	// One of the three policies: `Always`, `IfNotPresent`, `Never`
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
+
+	// Environment variables that will be used by CacheFS Fuse
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// Resources that will be requested by CacheFS Fuse.
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Options mount options that fuse pod will use
+	// +optional
+	Options map[string]string `json:"options,omitempty"`
+
+	// If the fuse client should be deployed in global mode,
+	// otherwise the affinity should be considered
+	// +optional
+	Global bool `json:"global,omitempty"`
```

Comment:
We don't support Global anymore. You can remove it.

Reply:
Removed, thanks!